jscomp/others/js_date.ml

```diff
@@ -0,0 +1,111 @@
+type t
+
+external valueOf : t -> float = "" [@@bs.send]
+(** returns the primitive value of this date, equivalent to getTime *)
+
+external make : unit -> t = "Date" [@@bs.new]
+(** returns a date representing the current time *)
+
+external fromFloat : float -> t = "Date" [@@bs.new]
+external fromString : string -> t = "Date" [@@bs.new]
+let fromString s =
+  let date = fromString s in
+  if valueOf date |> Js_float.isNaN |> Js.to_bool  then
+    None
+  else
+    Some date
+
+external makeWithYM : year:float -> month:float -> t = "Date" [@@bs.new]
+external makeWithYMD : year:float -> month:float -> date:float -> t = "Date" [@@bs.new]
+external makeWithYMDH : year:float -> month:float -> date:float -> hours:float -> t = "Date" [@@bs.new]
```

Comment:
In OCaml, it is recommended to have a `unit` separator when all arguments are labeled or optional.

```ocaml
external makeWithYM : year:float -> month:float -> unit -> t 
```